icu4j/main/tests/core/src/com/ibm/icu/dev/test/message2/Args.java

```diff
@@ -0,0 +1,187 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+package com.ibm.icu.dev.test.message2;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Convenience class that provides the same functionality as
+ * <code>Map.of</code> introduced in JDK 11, which can't be used for ICU4J.
```

Comment:
Question: Should the returned Map be unmodifiable? Maybe to prove that the formatter does not modify it?

Reply:
Done.
Updated the javadoc, and changed all returns to
`return Collections.unmodifiableMap(result);`